cloud-checksum/src/io/mod.rs

```diff
@@ -106,9 +106,25 @@ impl TryFrom<&str> for Provider {
     }
 }
 
-pub async fn default_s3_client() -> Result<Client> {
-    let config = load_defaults(BehaviorVersion::latest()).await;
-    Ok(Client::new(&config))
+pub async fn default_s3_client(make_anonymous: bool) -> Result<Client> {
+    let sdk_config = match make_anonymous {
+        true => {
+            aws_config::defaults(BehaviorVersion::v2025_01_17())
+                .region("ap-southeast-2")
```

Comment:
We may need to require a `--source-region` or something - because where there are no credentials - AWS still needs a region to attempt to hit for the s3 operations.